### PR TITLE
Add stored procedure test in mysql2

### PR DESCRIPTION
**Pull request by kamipo**

### CODE REVIEW

File: activerecord/test/cases/adapters/mysql/sp_test.rb [hunk 1]
```diff
@@ -1,15 +1,29 @@
...
+  def setup
+    @connection = ActiveRecord::Base.connection
+  end
+
   # Test that MySQL allows multiple results for stored procedures
-  if defined?(Mysql) && Mysql.const_defined?(:CLIENT_MULTI_RESULTS)
+  #
+  # In MySQL 5.6, CLIENT_MULTI_RESULTS is enabled by default.
+  # http://dev.mysql.com/doc/refman/5.6/en/call.html
+  if current_adapter?(:MysqlAdapter) &&
```
Comment (sgrif) [inline activerecord/test/cases/adapters/mysql/sp_test.rb]: You don't need to check the current adapter here.

File: activerecord/test/cases/adapters/mysql2/sp_test.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  def setup
+    @connection = ActiveRecord::Base.connection
+  end
+
+  # Test that MySQL allows multiple results for stored procedures
+  #
+  # In MySQL 5.6, CLIENT_MULTI_RESULTS is enabled by default.
+  # http://dev.mysql.com/doc/refman/5.6/en/call.html
+  if current_adapter?(:Mysql2Adapter) && ActiveRecord::Base.connection.send(:version) >= '5.6.0'
```
Comment (sgrif) [inline activerecord/test/cases/adapters/mysql2/sp_test.rb]: You don't need to check the current adapter here.

Comment (sgrif) [inline activerecord/test/cases/adapters/mysql2/sp_test.rb]: Let's just make `version` public rather than using `send`

**Comment by sgrif** [comment]
Can you add a test for a procedure that returns more than one row?

**Comment by kamipo** [comment]
Thank you for your review. I added a test for a procedure that returns more than one row!

File: activerecord/test/schema/mysql2_specific_schema.rb [hunk 3]
```diff
@@ -40,6 +40,17 @@
 END
 SQL
 
+  ActiveRecord::Base.connection.execute <<-SQL
+DROP PROCEDURE IF EXISTS topics;
+SQL
+
+  ActiveRecord::Base.connection.execute <<-SQL
```
Comment (sgrif) [inline activerecord/test/schema/mysql2_specific_schema.rb]: Could these be merged into a single command?

File: activerecord/test/schema/mysql2_specific_schema.rb [hunk 4]
```diff
@@ -40,6 +40,17 @@
...
+CREATE PROCEDURE topics(IN num INT) SQL SECURITY INVOKER
+BEGIN
+	select * from topics limit num;
+END
```
Comment (sgrif) [inline activerecord/test/schema/mysql2_specific_schema.rb]: Can you indent these one level further than the `execute` line?

**Comment by sgrif** [comment]
A few more nitpicks. You can squash after they're addressed.

File: activerecord/test/schema/mysql2_specific_schema.rb (hunk 3 above)
Comment (kamipo) [inline activerecord/test/schema/mysql2_specific_schema.rb]: We need to enable `CLIENT_MULTI_STATEMENTS` for multiple statements into a single command.
https://dev.mysql.com/doc/refman/5.6/en/c-api-multiple-queries.html

Should we enable the option for these queries?

**Pull request merged**